src/bubble_swarm/already_optimized.rs

```diff
@@ -0,0 +1,23 @@
+use crate::math2d::circle::Circle;
+
+pub struct AlreadyOptimizedCircles {
+    circles: Vec<Circle>,
+}
+
+impl AlreadyOptimizedCircles {
+    pub fn new() -> Self {
+        Self { circles: vec![] }
+    }
+
+    pub fn get_circles(&self) -> &Vec<Circle> {
+        &self.circles
+    }
+
+    pub fn add_circle(&mut self, circle: Circle) {
+        self.circles.push(circle);
+    }
+
+    pub fn area(&self) -> f32 {
+        self.circles.iter().map(|c| c.area()).sum()
```

Comment:
that is so cool, that you can express the area without a for loop in rust :)

Reply:
I like iterators a lot. I actually wanted to discuss this topic with you. Some potential refactoring of raw loops into iterators. They are zero-cost in Rust, they are chainable, and sometimes may provide some flexibility in more complex operations. Also, I just want to learn them better :)